lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -1611,3 +1630,25 @@ pub(crate) struct EmptyYieldVal;
 fn default_fatal_handler(inst: &Instance) -> ! {
     panic!("> instance {:p} had fatal error: {}", inst, inst.state);
 }
+
+/// Used by hosts to limit resource consumption of instances.
+///
+/// An instance can be configured with a resource limiter so that hosts can take into account non-WebAssembly
+/// resource usage to determine if linear memory should be permitted to grow.
+pub trait ResourceLimiter {
+    /// Called when the the instance's linear memory has been requested to grow.
+    ///
+    /// * `current` gives the current size of the linear memory, in bytes.
+    /// * `desired` gives the desired new size of the linear memory, in bytes.
+    /// * `limit` gives the configured maximum size of the linear memory, in bytes.
```

Comment:
I had to read through the call site above a couple times because this argument was kind of surprising to me. On a purely intuitive level, it feels like something called a `ResourceLimiter` shouldn't have to be told what its limit is. I wonder if this is a symptom of needing a more evocative name for this trait (perhaps `LimitEnforcer`?) or whether there's a different way of documenting its intended use.